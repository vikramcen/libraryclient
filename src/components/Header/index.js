import React, { useState, useEffect } from "react";
import Nav from '../../components/Nav';

export default function Header() {
  return (
    <header className="header d-flex justify-content-between px-4 align-items-center ">
      <div className="logo-container d-flex ">
        <a href="/">
          <img src="/logo1.png" alt="" style={{height: '60px'}}/>
        </a>
      </div>
      <Nav />
    </header>
  )
}