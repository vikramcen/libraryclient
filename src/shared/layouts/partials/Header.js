import { ROOT_PAGE_URL } from '@constants/navigator';
import mainLogo from './logo1.png';
//import mainLogo from '@images/logo1.png';
import { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { ReactSVG } from 'react-svg';
import CartDropdown from './CartDropdown';
import Navbar from './Navbar';
import Topbar from './Topbar';

const Header = () => {
  // get keyword from url
  const queryString = window.location.search;
  const urlParams = new URLSearchParams(queryString);
  const keyword = urlParams.get('keyword');
  const [search, setSearch] = useState(keyword || '');
  const navigate = useNavigate();

  const onClickSearch = () => {
    if (search) {
      navigate(`${ROOT_PAGE_URL.BOOKS}?keyword=${search}`);
    }else{
      navigate(`${ROOT_PAGE_URL.BOOKS}`);
    }
  };

  return (
    <header>
      <Topbar />
      <div className="header-mid-area py-4 ptb-40">
        <div className="container">
          <div className="row items-center">
            <div className="col-lg-3 col-md-5 col-12">
              <div className="header-search">
                <form action="#">
                  <input
                    type="text"
                    value={search}
                    onChange={(e) => setSearch(e.target.value)}
                    placeholder="Search entire store here..."
                  />
                  <button type="button" onClick={onClickSearch}>
                    <i className="fa fa-search" />
                  </button>
                </form>
              </div>
            </div>
            <div className="col-lg-6 col-md-4 col-12">
              <div className="logo-area text-center logo-xs-mrg flex items-center justify-center">
                <Link to={'/'}>
                  <img src={mainLogo} />
                </Link>
              </div>
            </div>
            <div className="col-lg-3 col-md-3 col-12">
              <CartDropdown />
            </div>
          </div>
        </div>
      </div>
      <Navbar />
    </header>
  );
};

export default Header;
