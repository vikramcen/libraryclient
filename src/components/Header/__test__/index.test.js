import React from "react";
import { render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

afterEach(cleanup);

describe('Header component', () => {
  // First Test
  it('renders', () => {
    render(<Header/>);
  });
  // Second Test
  it('matches snapshot DOM node structure', () => {
    //render About
    const {asFragment} = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  })
})