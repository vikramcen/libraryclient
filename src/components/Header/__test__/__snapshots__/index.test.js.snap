// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="header d-flex justify-content-between px-4 align-items-center "
  >
    <div
      class="logo-container d-flex "
    >
      <a
        href="/"
      >
        <img
          alt=""
          src="/logo1.png"
          style="height: 60px;"
        />
      </a>
    </div>
    <nav
      class="navbar navbar-expand-lg nav-hide"
    >
      <div
        class="nav-links mx-3"
      >
        <ul
          class="navbar-nav d-mob-none"
        >
          <li
            class="nav-item mx-2 width-max"
          >
            <a
              href="/"
            >
              HOME
            </a>
          </li>
          <li
            class="nav-item mx-2 width-max"
          >
            <a
              href="/catalog"
            >
              CATALOG
            </a>
          </li>
        </ul>
      </div>
      <div
        class="login-container d-mob-none"
      >
        <div
          class="d-flex"
        >
          <a
            class="btn btn-primary mx-2"
            href="/register"
          >
            REGISTER
          </a>
          <a
            class="btn btn-success mx-2"
            href="/login"
          >
            LOGIN
          </a>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="header d-flex justify-content-between px-4 align-items-center "
  >
    <div
      class="logo-container d-flex "
    >
      <a
        href="/"
      >
        <img
          alt=""
          src="/logo1.png"
          style="height: 60px;"
        />
      </a>
    </div>
    <nav
      class="navbar navbar-expand-lg nav-hide"
    >
      <div
        class="nav-links mx-3"
      >
        <ul
          class="navbar-nav d-mob-none"
        >
          <li
            class="nav-item mx-2 width-max"
          >
            <a
              href="/"
            >
              HOME
            </a>
          </li>
          <li
            class="nav-item mx-2 width-max"
          >
            <a
              href="/catalog"
            >
              CATALOG
            </a>
          </li>
        </ul>
      </div>
      <div
        class="login-container d-mob-none"
      >
        <div
          class="d-flex"
        >
          <a
            class="btn btn-primary mx-2"
            href="/register"
          >
            REGISTER
          </a>
          <a
            class="btn btn-success mx-2"
            href="/login"
          >
            LOGIN
          </a>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
